apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -56,127 +55,665 @@ struct arastorage_data_type_s {
 
-char g_query[QUERY_LENGTH];
-bool g_check;
-bool g_init;
+static char g_query[QUERY_LENGTH];
+static bool g_check;
+static bool g_init;
+
+/**
+* @testcase             deinit_db
+* @brief                To de-initialize database resources
+* @scenario             de-initialize database resources
+* @apicovered           db_deinit
+* @precondition         none
+* @postcondition        none
+*/
+static void deinit_db(void)
+{
+	if (g_init) {
+		db_deinit();
+	}
+}
+
+/**
+* @testcase             db_cleanup
+* @brief                To de-initialize database resources and to remove relation name from DB
+* @scenario             If used api returns DB_OK
+* @apicovered           db_exec
+* @precondition         none
+* @postcondition        none
+*/
+static void db_cleanup(const char *rel)
+{
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "REMOVE RELATION %s;", rel);
+	db_exec(g_query);
+}
+
+/**
+* @testcase             itc_arastorage_db_init_deinit_p
+* @brief                To check initialization and deinitialization database resources
+* @scenario             check initialization and deinitialization database operation returns DB_OK
+* @apicovered           db_deinit, db_init
+* @precondition         none
+* @postcondition        none
+*/
+void itc_arastorage_db_init_deinit_p(void)
+{
+	db_result_t ret;
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	ret = db_deinit();
+	TC_ASSERT_EQ("db_deinit", ret, DB_OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             itc_arastorage_db_init_deinit_p_multi_time
+* @brief                To check initialization and deinitialization database resources
+* @scenario             check iterative initialization and deinitialization database operation returns DB_OK
+* @apicovered           db_deinit, db_init
+* @precondition         none
+* @postcondition        none
+*/
+void itc_arastorage_db_init_deinit_p_multi_time(void)
+{
+	db_result_t ret;
+	int iteration = 0;
+
+	for (iteration = 0; iteration < 10; ++iteration) {
+		ret = db_init();
+		TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+		ret = db_deinit();
+		TC_ASSERT_EQ("db_deinit", ret, DB_OK);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             itc_arastorage_db_print_header_n_after_deinit
+* @brief                To check print of the related information : relation, attribute name
+* @scenario             If used api returns DB_OK
+* @apicovered           db_print_header, db_query
+* @precondition         Database resource must be deinitialized
+* @postcondition        none
+*/
+void itc_arastorage_db_print_header_n_after_deinit(void)
+{
+	db_result_t ret;
+	const char *rel_name = RELATION_NAME2;
+
+	ret = db_init();
+	TC_ASSERT_EQ("db_init", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
+	snprintf(g_query, QUERY_LENGTH, "CREATE RELATION %s;", rel_name);
+	ret = db_exec(g_query);
+	TC_ASSERT_EQ("db_exec", ret, DB_OK);
+
+	memset(g_query, 0, QUERY_LENGTH);
```

Comment:
Here also create_attribute() will be fine

Reply:
I mean you can make one function for below.
memset(g_query, 0, QUERY_LENGTH);
snprintf(g_query, QUERY_LENGTH, "CREATE ATTRIBUTE %s DOMAIN long IN %s;", g_attribute_set[1], rel_name);
ret = db_exec(g_query);
if (ret != DB_OK) {
	error_exec(rel_name);
	return;
}